File: consensus/beacon/src/main/java/org/hyperledger/besu/consensus/beacon/containers/misc/BeaconBlockHeader.java
package org.hyperledger.besu.consensus.beacon.containers.misc;

import org.hyperledger.besu.consensus.beacon.types.Root;
import org.hyperledger.besu.consensus.beacon.types.Slot;
import org.hyperledger.besu.consensus.beacon.types.ValidatorIndex;

public record BeaconBlockHeader(
        Slot slot,
        ValidatorIndex proposerIndex,
        Root parentRoot,
        Root stateRoot,
        Root bodyRoot
) {
}

File: beacon/spec/src/main/kotlin/maru/beacon/spec/BeaconBlockHeader.kt
package maru.beacon.spec

import org.apache.tuweni.bytes.Bytes32
import org.hyperledger.besu.consensus.common.bft.ConsensusRoundIdentifier

interface BeaconBlockHeader {
  val consensusRoundIdentifier: ConsensusRoundIdentifier
  val proposer: Validator
  val parentRoot: Bytes32
  val stateRoot: Bytes32
  val bodyRoot: Bytes32
}
